airflow-core/src/airflow/ui/src/queries/useEditConnection.tsx

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { useQueryClient } from "@tanstack/react-query";
+import { useState } from "react";
+
+import { useConnectionServiceGetConnectionsKey, useConnectionServicePatchConnection } from "openapi/queries";
+import { toaster } from "src/components/ui";
+import type { ConnectionBody } from "src/pages/Connections/AddConnectionButton";
+
+export const useEditConnection = (
+  initialConnection: ConnectionBody,
+  {
+    onSuccessConfirm,
+  }: {
+    onSuccessConfirm: () => void;
+  },
+) => {
+  const queryClient = useQueryClient();
+  const [error, setError] = useState<unknown>(undefined);
+
+  const onSuccess = async () => {
+    await queryClient.invalidateQueries({
+      queryKey: [useConnectionServiceGetConnectionsKey],
+    });
+
+    toaster.create({
+      description: "Connection has been edited successfully",
+      title: "Connection Edit Request Submitted",
+      type: "success",
+    });
+
+    onSuccessConfirm();
+  };
+
+  const onError = (_error: unknown) => {
+    setError(_error);
+  };
+
+  const { isPending, mutate } = useConnectionServicePatchConnection({
+    onError,
+    onSuccess,
+  });
+
+  const editConnection = (requestBody: ConnectionBody) => {
+    const updateMask: Array<string> = [];
+
+    if (requestBody.conf !== initialConnection.conf) {
+      updateMask.push("conf");
```

Comment:
This needs to be "extra" to be working.
```suggestion
      updateMask.push("extra");
```

Reply:
Does it maybe make sense for consistency to rename the field `ConnectionBody` as well to be `extra`?

Reply:
Ohh nice, good catch!
Yes, This makes sense, let me change all together .